Storybook: Add stories for the TextAlignmentControl component

himanshupathak95 opened the pull request:
Part of https://github.com/WordPress/gutenberg/issues/67165

## What?
This PR improves the Storybook stories for the TextAlignmentControl component in the block editor. It includes comprehensive stories demonstrating different text alignment states.

## Why?
As part of the ongoing effort to improve component documentation and testing (tracked in #22891), we need comprehensive stories for all Block Editor components. This PR helps users and developers better understand the TextAlignmentControl component's capabilities and usage.

## Testing Instructions
- Start Storybook by running `npm run storybook:dev`
- Open Storybook at http://localhost:50240/
- Navigate to "BlockEditor/TextAlignmentControl" in the Storybook sidebar
- Test the following stories:
    - Default (left alignment)
    - WithLeftAlignment (explicit left alignment)
    - WithCenterAlignment (center alignment)
    - WithRightAlignment (right alignment)
    - WithJustifyAlignment (justify alignment)
    - WithCustomClass (custom class name)

## Screencast



https://github.com/user-attachments/assets/80a0b990-6c2b-4224-80cd-e5aae11eec2d

himanshupathak95 committed:
Storybook: Add stories for the text-alignment-control component

t-hamano commented on packages/block-editor/src/components/text-alignment-control/stories/index.story.js:
```diff
@@ -1,39 +1,162 @@
+			if ( onChange ) {
+				onChange( newValue );
+			}
+		};
```
It could be made simpler without using the `useEffect` Hook. Take a look at this code:

https://github.com/WordPress/gutenberg/blob/680fef531408601dcd04e199aefdfac7190cb577/packages/block-editor/src/components/alignment-control/stories/index.story.js#L37-L51

t-hamano commented on packages/block-editor/src/components/text-alignment-control/stories/index.story.js:
```diff
@@ -1,39 +1,162 @@
+ */
 export default {
 	title: 'BlockEditor/TextAlignmentControl',
 	component: TextAlignmentControl,
```
```suggestion
	component: TextAlignmentControl,
	parameters: {
		docs: {
			canvas: { sourceState: 'shown' },
			description: {
				component: 'Control to facilitate text alignment selections.',
			},
		},
	},
```

Let's provide component description and source code.

t-hamano commented on packages/block-editor/src/components/text-alignment-control/stories/index.story.js:
```diff
@@ -1,39 +1,162 @@
+				},
+				defaultValue: { summary: 'left' },
+			},
+		},
```
```suggestion
		value: {
			control: { type: null },
			description: 'Currently selected text alignment value.',
			table: {
				type: {
					summary: 'string',
				},
			},
		},
```

t-hamano commented on packages/block-editor/src/components/text-alignment-control/stories/index.story.js:
```diff
@@ -1,39 +1,162 @@
 		options: {
-			control: 'check',
-			options: [ 'left', 'center', 'right', 'justify' ],
+			control: 'array',
```
Is there a reason you changed the control? Personally, I think the checkboxes are easier to understand:

![image](https://github.com/user-attachments/assets/aa8312ba-e9e3-4385-a7d4-c89fcd7323e9)

t-hamano reviewed (commented):
Thanks for the PR!

Both the `value` prop and the `classname` prop can be changed in a story, so splitting them all out as stories seems a bit redundant 😅

Can't we just leave them as one story?

himanshupathak95 committed:
Storybook: Update TextAlignmentControl story to follow best practices and simplify the structure

himanshupathak95 committed:
Merge branch 'trunk' into storybook/text-alignment-control

himanshupathak95 commented:
Thanks, @t-hamano for the wonderful suggestions.

I have addressed all the feedback and created a screencast of the new changes. Please let me if it tests well for you. 



https://github.com/user-attachments/assets/2e6cf48b-2a95-4de3-a742-356c901bca73

t-hamano commented on packages/block-editor/src/components/text-alignment-control/stories/index.story.js:
```diff
@@ -8,32 +8,84 @@ import { useState } from '@wordpress/element';
 			control: 'check',
-			options: [ 'left', 'center', 'right', 'justify' ],
+			description: 'Array of text alignment options to display.',
+			options: ALIGNMENT_OPTIONS,
```
```suggestion
			options: [ 'left', 'center', 'right', 'justify' ],
```

The `ALIGNMENT_OPTIONS` variable will only be used once, so let's hard-code it here.

t-hamano commented on packages/block-editor/src/components/text-alignment-control/stories/index.story.js:
```diff
@@ -8,32 +8,84 @@ import { useState } from '@wordpress/element';
+	args: {
+		value: 'left',
+		options: ALIGNMENT_OPTIONS,
+	},
```
This should be able to be deleted.

t-hamano commented on packages/block-editor/src/components/text-alignment-control/stories/index.story.js:
```diff
@@ -8,32 +8,84 @@ import { useState } from '@wordpress/element';
+/**
+ * TextAlignmentControl Properties
+ */
+const meta = {
```
```suggestion
const meta = {
```

t-hamano reviewed (commented):
Thanks for the update! I left some small suggestions at the end, but it works fine 👍

himanshupathak95 committed:
Storybook: Simplify TextAlignmentControl story

himanshupathak95 commented:
Thanks again for the follow-ups @t-hamano! 

The suggestions help simplify the structure further. I have addressed all the feedback 👍🏻

t-hamano reviewed (approved):
LGTM! Thank you fo your work 👍

t-hamano commented on packages/block-editor/src/components/text-alignment-control/stories/index.story.js:
```diff
@@ -8,32 +8,72 @@ import { useState } from '@wordpress/element';
+export default meta;
 
-export const Default = Template.bind( {} );
+/**
+ * Default story showing TextAlignmentControl with left alignment
+ */
+export const Default = {
```
Nit: There is only one story, so I don't think this comment is necessary anymore.

t-hamano commented on packages/block-editor/src/components/text-alignment-control/stories/index.story.js:
```diff
@@ -8,32 +8,72 @@ import { useState } from '@wordpress/element';
+export default meta;
 
-export const Default = Template.bind( {} );
+/**
+ * Default story showing TextAlignmentControl with left alignment
+ */
+export const Default = {
```
```suggestion
export const Default = {
```

Nit: There is only one story, so I don't think this comment is necessary anymore.

himanshupathak95 replied:
Thanks for the approval @t-hamano 🎉 

I have removed the comment.

himanshupathak95 committed:
Storybook: Simplify the documentation for TextAlignmentControl story

himanshupathak95 committed:
Merge branch 'trunk' into storybook/text-alignment-control
